crates/ra_lsp_server/src/main_loop/handlers.rs

```diff
@@ -525,6 +525,18 @@ pub fn handle_hover(
             contents: HoverContents::Scalar(MarkedString::String(result.join("\n\n---\n"))),
             range: Some(range),
         }));
+    } else {
+        let file_id = params.text_document.try_conv_with(&world)?;
```

Comment:
Note that we have `world.analysis().approximately_resolve_symbol(position)?` above, which means that if we fail to resove symbol (if, for example, the cursor is on something which is not an identifier, like `+`), we return `None`. 

I suggest refactoring impl into two functions: `get_doc() -> Option<String>` and `get_type() -> Option<String>` and call then from the main handler, so as to avoid accidental `?` problelms